job-server-api/src/spark.jobserver/NamedObjectSupport.scala

```diff
@@ -0,0 +1,150 @@
+package spark.jobserver
+
+import akka.util.Timeout
+import java.lang.ThreadLocal
+import java.util.concurrent.atomic.AtomicReference
+
+import scala.concurrent.duration.Duration
+
+trait NamedObject
+
+/**
+ * implementations of this abstract class should handle the specifics
+ * of each named object's persistence
+ */
+abstract class NamedObjectPersister[O <: NamedObject] {
+
+  /**
+   * persists an object with the given name in
+   * the context
+  //TODO what is the context needed for? c: ContextLike,
+   * @param c - context to store the object in
+   * @param name - name of the object
+   */
+  def saveToContext(namedObj: O, name: String): Unit
+
+  //TODO - what is this for? Is this comment correct?
+  /**
+   * tries to retrieve an object with the given name from
+   * the context so that is can be stored in the list of named objects
+   * @param c - context to retrieve the object from
+   * @param name - name of the object
+   * @return the named object with the given name. TODO - wouldn't it be better to
+   * return Option[O] ?
+   */
+  //def getFromContext(c: ContextLike, name: String): O
+
+  /**
+   * update reference to named object so that it does not get GCed
+   */
+  def refresh(namedObject: O): O
+
+  //TODO - what about 'unpersist' / removeFromContext / destroy?
+  //proposal:
+  // def removeFromContext(c: ContextLike, name: String): Unit
```

Comment:
I do believe that we need a `destroy` method so that the persister knowns when to unpersist the named object.